apps/web/src/features/safenet/components/new-safe/SafenetReview/index.tsx

```diff
@@ -0,0 +1,151 @@
+import css from './styles.module.css'
+
+export type UseSubmitSafenetReviewHandlerProps = {
+  data: NewSafeFormData
+  onSubmit: (data: Partial<NewSafeFormData>) => void
+  setSubmitError: Dispatch<SetStateAction<string | undefined>>
+  setIsCreating: Dispatch<SetStateAction<boolean>>
+}
+
+export type SafenetReviewType = {
+  handleCreateSafeClick: () => void
+}
+
+export const useSubmitSafenetReviewHandler = ({
+  data,
+  onSubmit,
+  setSubmitError,
+  setIsCreating,
+}: UseSubmitSafenetReviewHandlerProps): SafenetReviewType => {
+  const router = useRouter()
+  const dispatch = useAppDispatch()
+  const [deploy, { data: deploymentData, error }] = useLazyDeploySafenetAccountQuery()
+
+  useEffect(() => {
+    if (!error) return
+    console.error(error)
+    setSubmitError('Error creating the Safe Account. Please try again later.')
+    setIsCreating(false)
+  }, [error, setSubmitError, setIsCreating])
+
+  useEffect(() => {
+    if (!deploymentData || !data) return
+
+    const replayedSafeWithNonce: ReplayedSafeProps = {
+      factoryAddress: deploymentData.factoryAddress,
+      masterCopy: deploymentData.masterCopy,
+      safeAccountConfig: {
+        threshold: deploymentData.safeAccountConfig.threshold,
+        owners: data.owners.map((owner) => owner.address),
+        to: deploymentData.safeAccountConfig.to || ZERO_ADDRESS,
+        data: deploymentData.safeAccountConfig.data || EMPTY_DATA,
+        fallbackHandler: deploymentData.safeAccountConfig.fallbackHandler || ZERO_ADDRESS,
+        paymentReceiver: ZERO_ADDRESS, // data.paymentReceiver ?? ECOSYSTEM_ID_ADDRESS,
+      },
+      saltNonce: deploymentData.saltNonce,
+      safeVersion: deploymentData.safeVersion,
+    }
+
+    data.networks.map((chain) => {
+      gtmSetChainId(chain.chainId)
+
+      replayCounterfactualSafeDeployment(
+        chain.chainId,
+        deploymentData.safeAddress,
+        replayedSafeWithNonce,
+        data.name,
+        dispatch,
+        PayMethod.PayNow,
+      )
+
+      safeCreationDispatch(SafeCreationEvent.RELAYING, {
+        groupKey: CF_TX_GROUP_KEY,
+        safeAddress: deploymentData.safeAddress,
+        chainId: chain.chainId,
+      })
+
+      //trackEvent({ ...OVERVIEW_EVENTS.PROCEED_WITH_TX, label: 'deployment', category: CREATE_SAFE_CATEGORY })
+      //trackEvent(CREATE_SAFE_EVENTS.SUBMIT_CREATE_SAFE)
+    })
+
+    onSubmit(data)
+
+    // Update addressbook with owners and Safe on all chosen networks
+    dispatch(
+      updateAddressBook(
+        data.networks.map((network) => network.chainId),
+        deploymentData.safeAddress,
+        data.name,
+        data.owners,
+        data.threshold,
+      ),
+    )
+
+    router?.push({
+      pathname: AppRoutes.home,
+      query: { safe: `${data.networks[0].shortName}:${deploymentData.safeAddress}` },
+    })
+  }, [data, deploymentData, dispatch, onSubmit, router])
+
+  const handleCreateSafeClick = async () => {
+    setIsCreating(true)
+
+    deploy({
```

Comment:
Instead of handling the result processing through `useEffect(func, [deploymentData])` we could also do 
```ts
   const {data: deploymentData} = await deploy()
   // Process deploymentData.
```

This way we have less side-effects handling.